File: debot/src/main/java/com/tomoima/debot/DebotConfigurator.java
package com.tomoima.debot;

import android.content.Context;

import com.tomoima.debot.strategy.CheckAppVersionStrategy;
import com.tomoima.debot.strategy.CheckDpiStrategy;
import com.tomoima.debot.strategy.DebotStrategy;
import com.tomoima.debot.strategy.DebotCallActivityMethodStrategy;
import com.tomoima.debot.strategy.ShowActivityInfoStrategy;

import java.util.ArrayList;

public class DebotConfigurator {

    public static void configureWithDefault(Context context) {
        new DebotStrategies().initialize(createDefaultMenuConfig(context));
    }

    public static void configureWithCustomizedMenu(Context context, ArrayList<DebotStrategy> customList){
        ArrayList<DebotStrategy> registerList = createDefaultMenuConfig(context);
        registerList.addAll(customList);
        new DebotStrategies().initialize(registerList);
    }

    private static ArrayList<DebotStrategy> createDefaultMenuConfig(Context context){
        DebotStrategyBuilder builder = new DebotStrategyBuilder.Builder(context)
                .registerMenu("check dpi", new CheckDpiStrategy())
                .registerMenu("showDebugMenu intent", new ShowActivityInfoStrategy())
                .registerMenu("check App ver", new CheckAppVersionStrategy())
                .registerMenu("dev input", new DebotCallActivityMethodStrategy("debugInput"))
                .build();
        return builder.getStrategyList();
    }
}

File: debot/src/main/java/com/tomoima/debot/DebotConfigurator.kt
package com.tomoima.debot

import android.content.Context
import com.tomoima.debot.strategy.*
import java.util.*

object DebotConfigurator {

    @JvmStatic fun configureWithDefault() {
        DebotStrategies.initialize(createDefaultMenuConfig())
    }

    @JvmStatic fun configureWithCustomizedMenu(customList:
    ArrayList<DebotStrategy>) {
        val registerList = createDefaultMenuConfig()
        registerList.addAll(customList)
        DebotStrategies.initialize(registerList)
    }

    private fun createDefaultMenuConfig(): ArrayList<DebotStrategy> {
        val builder = DebotStrategyBuilder.Builder()
                .registerMenu("check dpi", CheckDpiStrategy())
                .registerMenu("showDebugMenu intent", ShowActivityInfoStrategy())
                .registerMenu("check App ver", CheckAppVersionStrategy())
                .registerMenu("dev input", DebotCallActivityMethodStrategy("debugInput"))
                .build()
        return builder.strategyList
    }
}